test/src/test/resources/fixtures/hydration/basic-conditional-hydration-with-true-condition.yml

```
@@ -0,0 +1,129 @@
+name: "basic conditional hydration with true condition"
+enabled: true
+overallSchema:
+  service2: |
+    type Query {
+      barById(id: ID): Bar
+    }
+    type Bar {
+      id: ID
+      name: String
+      type: String
+    }
+  service1: |
+    type Query {
+      foo: Foo
+    }
+    type Foo {
+      id: ID
+      type: String
+      bar: Bar @hydrated(
+        service: "service2" 
+        field: "barById" 
+        arguments: [
+          {
+            name: "id" 
+            value: "$source.barId"
+          }
+        ]
+        when: {
+          result: {
+              sourceField: "type"
+              predicate: { equals: "someType" }
+          }
+        }
+      )
+      barLongerInput: Bar @hydrated(service: "service2" field: "barById" arguments: [{name: "id" value: "$source.fooDetails.externalBarId"}])
+    }
+underlyingSchema:
+  service2: |
+    type Bar {
+      id: ID
+      name: String
+      type: String
+    }
+    
+    type Query {
+      barById(id: ID): Bar
+    }
+  service1: |
+    type Foo {
+      barId: ID
+      fooDetails: FooDetails
+      id: ID
+    type: String
+    }
+    
+    type FooDetails {
+      externalBarId: ID
+    }
+    
+    type Query {
+      foo: Foo
+    }
+query: |
+  query {
+    foo {
+      bar {
+        name
+      }
+    }
+  }
+variables: { }
+serviceCalls:
+  - serviceName: "service1"
+    request:
+      query: |
+        query {
+          foo {
+            __typename__hydration__bar: __typename
+            hydration__bar__barId: barId
+            hydration__bar__type: type
+          }
+        }
+      variables: { }
+    # language=JSON
+    response: |-
+      {
+        "data": {
+          "foo": {
+            "hydration__bar__barId": "barId",
+            "__typename__hydration__bar": "Foo",
+            "hydration__bar__type": "someType"
+          }
+        },
+        "extensions": {}
+      }
+  - serviceName: "service2"
+    request:
+      query: |
+        query {
+          barById(id: "barId") {
+            name
+          }
+        }
+      variables: { }
+    # language=JSON
+    response: |-
+      {
+        "data": {
+          "barById": {
+            "name": "Bar1",
+            "type": "someType"
```

```suggestion
            "name": "Bar1"
```

The query never asks for  `type`